backend/app/api/src/endpoints/global/platform/PatchPlatformEnpoint.ts

```diff
@@ -29,65 +36,100 @@ export class PatchPlatformEndpoint extends Endpoint<Params, Query, Body, Respons
     }
 
     async handle(request: DecodedRequest<Params, Query, Body>) {
-        await Context.authenticate()
+        await Context.authenticate();
 
         // Fast throw first (more in depth checking for patches later)
         if (!Context.auth.hasPlatformFullAccess()) {
-            throw Context.auth.error()
+            throw Context.auth.error();
         }
 
-        const platform = await Platform.getShared()
+        const platform = await Platform.getShared();
 
         if (request.body.privateConfig) {
             // Did we patch roles?
             if (request.body.privateConfig.roles) {
                 if (!Context.auth.canManagePlatformAdmins()) {
-                    throw Context.auth.error()
+                    throw Context.auth.error();
                 }
 
                 // Update roles
-                platform.privateConfig.roles = patchObject(platform.privateConfig.roles, request.body.privateConfig.roles)
+                platform.privateConfig.roles = patchObject(
+                    platform.privateConfig.roles,
+                    request.body.privateConfig.roles
+                );
             }
 
             if (request.body.privateConfig.emails) {
                 if (!Context.auth.hasPlatformFullAccess()) {
-                    throw Context.auth.error()
+                    throw Context.auth.error();
                 }
 
                 // Update roles
-                platform.privateConfig.emails = patchObject(platform.privateConfig.emails, request.body.privateConfig.emails)
+                platform.privateConfig.emails = patchObject(
+                    platform.privateConfig.emails,
+                    request.body.privateConfig.emails
+                );
             }
         }
 
         if (request.body.config) {
             if (!Context.auth.hasPlatformFullAccess()) {
-                throw Context.auth.error()
+                throw Context.auth.error();
             }
 
+            const newConfig = request.body.config;
+
             // Update config
-            if(request.body.config.premiseTypes) {
-                const oldConfig = platform.config.clone();
-                platform.config = patchObject(platform.config, request.body.config);
-                const newPremiseTypes = platform.config.premiseTypes;
-
-                // update setup step premise types
-                if(this.shouldUpdateSetupStepPremise(newPremiseTypes, oldConfig.premiseTypes)) {
-                    await SetupStepUpdater.updateSetupStepsForAllOrganizationsInCurrentPeriod({premiseTypes: newPremiseTypes});
+            if (newConfig) {
+                if (newConfig.premiseTypes || newConfig.responsibilities) {
+                    const oldConfig = platform.config.clone();
+                    platform.config = patchObject(platform.config, newConfig);
+                    const currentConfig = platform.config;
+
+                    this.updateSetupSteps(
```

Comment:
Zie nu wel dat we de steps hier al aanpassen nog voor we de platform hebben opgeslagen in de database, dat kan wel voor problemen zorgen